src/lib/input/input.md

```diff
@@ -129,18 +129,18 @@ the error messages.
 function myErrorStateMatcher(control: FormControl, form: FormGroupDirective | NgForm): boolean {
   // Error when invalid control is dirty, touched, or submitted
   const isSubmitted = form && form.submitted;
-  return !!(control.invalid && (control.dirty || control.touched || isSubmitted)));
+  return !!(control.invalid && (control.dirty || control.touched || isSubmitted));
```

Comment:
It has an effect because `isSubmitted` is potentially `undefined` (if `form` is `undefined`)